src/Symfony/Bridge/Twig/Translation/TwigExtractor.php

```diff
@@ -52,16 +52,22 @@ public function __construct(\Twig_Environment $twig)
     /**
      * {@inheritdoc}
      */
-    public function extract($directory, MessageCatalogue $catalogue)
+    public function extract($resource, MessageCatalogue $catalogue)
     {
-        // load any existing translation files
-        $finder = new Finder();
-        $files = $finder->files()->name('*.twig')->sortByName()->in($directory);
+        if (is_array($resource)) {
+            $files = array_filter($resource, function ($filename) {
```

Comment:
Some concerns about your comment:

- Isn't the `is_file` check too expensive if the array contains a lot of files?
- The second if statement is making this behave differently than before. Files with another extensions were ignored, no exception was thrown. I'd prefer to keep the old behaviour.
- Are we creating SplFileInfo instances only to keep the code from below the same?